platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/TimeLimitedMatcherFactory.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.util;
+
+import java.time.Duration;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Factory class used to generate time-limited Matcher instances.
+ *
+ * <p>If the regex runs for too long, this matcher will throw a RuntimeException - specifically a
+ * {@link RegExpTimeoutException} - that will throw if in use after their timeout. It is incumbent
+ * upon the caller so handle the runtime exception.
+ *
+ * <p>Example usage:
+ *
+ * <pre>
+ *     try {
+ *       Matcher matcher =
+ *           TimeLimitedMatcherFactory.matcher(
+ *               Pattern.compile(".*(this string).*"), "search this string for 'this string'");
+ *       if (matcher.matches()) {
+ *         // do something with the match
+ *       }
+ *     } catch (TimeLimitedMatcherFactory.RegExpTimeoutException e) {
```

Comment:
Since the exception is already scoped within the factory class, should we shorten its name to simply `TimeoutException`?

Reply:
Sure.